app/helpers/routes/findSelectedCatalogueItemInSession.js

```diff
@@ -0,0 +1,23 @@
+import { logger } from '../../logger';
+
+const findByType = {
+  additionalServices: catalogueItem => catalogueItem.additionalServiceId,
+  solutions: catalogueItem => catalogueItem.catalogueItemId,
+};
+
+export const findSelectedCatalogueItemInSession = ({
+  req, selectedItemId, sessionManager, catalogueItemsKey,
+}) => {
+  const additionalServices = sessionManager.getFromSession({ req, key: catalogueItemsKey });
+  const findCallback = catalogueItem => (
+    findByType[catalogueItemsKey](catalogueItem) === selectedItemId
+  );
+  const selectedItem = additionalServices.find(findCallback);
+
+  if (!selectedItem) {
+    logger.error(`Unable to find selected item ${selectedItemId} in session`);
+    throw new Error();

```

Comment:
Just suggestions to think about not saying change it...
Is it really an exceptional case to not find the selected item in session? How does the calling code know what to do with this error? Can it recover from it?

My thinking here is I want to check an item is in session & if not then I would then go to the API to retrieve it. An undefined check could satisfy this.

Reply:
this one is a little tricky. We would have to know how to handle this and which endpoint to call again to get the list of catalogueItems again. I think what this is doing here is fine for now. The session has expired or for whatever reason it can't find the selectedItem it will show the general error page and they'll have to start the journey again. To try and mitigate this would add a fair bit of complexity to the code.